akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -388,4 +405,60 @@ private[akka] class PersistentShard(
     }
   }
 
+  private def retrieveEntityRecoveryConfig(): Config = {
+    val path = settings.tuningParameters.entityRecoveryStrategyConfigPath
+    if (path.isEmpty)
+      ConfigFactory.empty()
+    else
+      context.system.settings.config.getConfig(path)
+  }
+}
+
+abstract class EntityRecoveryStrategy {
+  import ShardRegion.EntityId
+  import akka.actor.ActorContext
+  def recoverEntities(context: ActorContext, entities: Set[EntityId])
```

Comment:
Yes exactly. Let's not overcomplicate (explode the number of options people *could* set) for now. The two strategies make sense